app/models/location.rb

```diff
@@ -0,0 +1,13 @@
+class Location < ApplicationRecord
+
+  # Associations
+  has_one :address, dependent: :destroy
+  has_many :deals_locations
+  has_many :deals, through: :deals_locations
```

Comment:
dependent?

let's add a validation that if a location has even a single active deal then it can't be deleted